shtdlib.sh

```diff
@@ -204,17 +204,85 @@ function readlink_m {
     fi
 }
 
+# A function to get the status of shell options supported by 'set'
+# Outputs: 'on', 'off' or nothing for invalid/empty option names
+function get_sh_opts {
+    if [ -n "${1}" ]; then
+        shopt -o "${1}" 2> /dev/null | awk '{print $2}'
+    fi
+}
+
+
+# A function to toggle shell options in bash
+#  - Param 1: [REQUIRED] name of the option (from the 'set' builtin)
+#  - Param 2: [OPTIONAL] 'enable', 'disable' or empty/unset. If empty/unset,
```

Comment:
Consider using true/false evaluation rather than enable/disable, not sure if it's better but it would allow passing in functions or commands to determine if an option should be toggled and the shell built ins "true" "false" which of course evaluate to true and false.

Reply:
K